test/unit/test_util_math.cpp

```diff
@@ -168,7 +169,7 @@ TYPED_TEST(MathUtilTest, ptrdiff_t_Arithmetic_Mul) {
 
   TypeParam a, b;
 
-  if (type_size < arithmetic_size) {
+  if (type_size * 2 < arithmetic_size) {
     a = std::numeric_limits<TypeParam>::max();
     b = std::numeric_limits<TypeParam>::max();
   }
```

Comment:
I checked that the overflows reported by ubsan do indeed make sense, but please have a look to see if you think the tests are still testing what they should be testing.